lua/nvim-tree.lua

```diff
@@ -439,6 +447,10 @@ function M.setup(conf)
     _config.update_to_buf_dir = opts.update_to_buf_dir
   end
 
+	if opts.hide_root_folder then
+		view.View.hide_root_folder = true
```

Comment:
if this is a view option, add it to the view config and set it inside the view setup.